src/coreclr/jit/lclvars.cpp

```diff
@@ -2959,6 +2959,9 @@ void Compiler::lvaSetStruct(unsigned varNum, ClassLayout* layout, bool unsafeVal
         assert(layout->IsBlockLayout() || (layout->GetSize() != 0));
     }
 
+    unsigned classAttribs = info.compCompHnd->getClassAttribs(layout->GetClassHandle());
+    varDsc->SetIsSpan(classAttribs & CORINFO_FLG_SPAN);
```

Comment:
This should be inside the `!layout->IsBlockLayout()` check (block layouts do not have class handles)

Reply:
I should write this all down somewhere. Thank you.

Reply:
My main was old and I didn't see the call was under a condition like you've said. :(